avm/res/sql/server/tests/e2e/failover-group/main.test.bicep

```diff
@@ -0,0 +1,147 @@
+targetScope = 'subscription'
+
+metadata name = 'Using failover groups'
+metadata description = 'This instance deploys the module with failover groups.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-sql.servers-${serviceShort}-rg'
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'ssfog'
+
+@description('Optional. The password to leverage for the login.')
+@secure()
+param password string = newGuid()
+
+@description('Optional. A token to inject into the name of each resource. This value can be automatically injected by the CI.')
+param namePrefix string = '#_namePrefix_#'
+
+// Use paired regions
+// https://learn.microsoft.com/en-us/azure/reliability/cross-region-replication-azure
+var locationPrimary = 'eastasia'
+var locationSecondary = 'southeastasia'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: locationPrimary
+}
+
+// Create a secondary server for the failover group
+var secondaryServerName = '${namePrefix}${serviceShort}002'
```

Comment:
OK, I see how it works, thanks

Reply:
Thanks @peterbud,
last thing (in alignment with other AVM modules): Please remove the variable which is only used in one location into the `nestedDeployment` block:
```bicep
module nestedDependencies 'dependencies.bicep' = {
  scope: resourceGroup
  name: '${uniqueString(deployment().name, locationSecondary)}-nestedDependencies'
  params: {
    serverName: '${namePrefix}${serviceShort}002'
    location: locationSecondary
(...)
```

Other than that, I think we're good 💪 Maybe also run the `Set-AVMModule` script a last time - just in case 😄